champ/node/wallets/manager.rs

```diff
@@ -0,0 +1,291 @@
+use crate::config::{read_file, write_file};
+use crate::{config::WalletManagerConfig, state::ChampStateArc};
+use anyhow::Result;
+use lulw;
+use serde_json;
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+
+#[derive(Debug, Clone)]
+pub struct Wallet {
+    pub name: UserName,
+    pub locked: bool,
+    pub account_address: String,
+    pub authorized_roles: Vec<String>, // users with this role can access this account using our api
```

Comment:
I think it's fine if we remove this field for now as it's unused anyways, a neater solution would be to have wallet specific permissions which would make both giving roles and even specific users directly access possible. But this can all be done on the gRPC side later